### PR TITLE
Generate examples from schemas.

### PR DESCRIPTION
**Pull request by michaelgrahamevans**
- [x] Basic types
- [x] Enums
- [x] oneOf, anyOf, allOf
- [x] Any type
- [x] minimum, maximum, exclusiveMinimum and exclusiveMaximum
- [x] minLength and maxLength
- [x] Multi-type arrays
- [x] Array length

### CODE REVIEW

File: sphinxcontrib/openapi/renderers/_httpdomain.py [hunk 1]
```diff
@@ -315,3 +336,118 @@ def render_response_content(self, media_type, status_code):
...
+    >>> example = _generate_example_from_schema(schema)
+    >>> assert example == {
+    ...     "id": 1,
+    ...     "name": "John Smith",
+    ...     "tag": "string"
+    ... }
+    """
+    # If an example was provided then we use that
+    if "example" in schema:
+        return schema["example"]
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi/renderers/_httpdomain.py]: I think this stuff is covered by the renderer itself, when examples are probed in order:

* `examples`
* `example`
* `schema.example`
* *generate-example-from-schema*

That said, I think we don't need this check here and I'd prefer the function to be simply: get schema as input, generate example as output.

File: sphinxcontrib/openapi/renderers/_httpdomain.py [hunk 2]
```diff
@@ -315,3 +336,118 @@ def render_response_content(self, media_type, status_code):
             yield f"   Content-Type: {content_type}"
             yield f""
             yield from indented(example.splitlines())
+
+
+def _generate_example_from_schema(schema):
```
Comment (ikalnytskyi) [inline sphinxcontrib/openapi/renderers/_httpdomain.py]: I think this function might be useful alone and is not generally tied to this project. I'd suggest to put the whole function it some sub-module, let's say `sphinxcontrib/openapi/jsonschemaext.py`. Who knows? Maybe one day this goodie become a separate library or will be incorporated by another.

**Comment by ikalnytskyi** [comment]
@michaelgrahamevans Thanks for the work and your help! I'm so happy that you picked this up 'cause it's actually one of not many things I personally didn't want to work on. :) May I ask you to move and isolate the example generation code in some module within the package? I feel like it should be an internal library that can be used by any other renderer we may have in the future.

File: sphinxcontrib/openapi/renderers/_httpdomain.py (hunk 1 above)
Comment (michaelgrahamevans) [reply sphinxcontrib/openapi/renderers/_httpdomain.py]: I cannot find a clear description of it in the OpenAPI spec but the [Swagger docs](https://swagger.io/docs/specification/adding-examples/) (section called "Object and Property Examples") mention that individual object schema properties can have examples. I would like to keep this as it allows a user to partially override a generated example if what would have been generated is not appropriate. Would you mind if I keep this check but I move it into the loop over object properties?

Comment (ikalnytskyi) [reply sphinxcontrib/openapi/renderers/_httpdomain.py]: Makes sense. Let's keep in then.

**Comment by marschmidt89** [comment]
Do you plan to merge the feature anytime soon? 

This is my first time dealing with Sphinx and the created API documentation looks really good, in fact better than with most other tools. The only problem is that the schemas and sample responses are missing. Means the feature would really be extremely helpful :)

**Comment by ikalnytskyi** [comment]
Awesome! Thank you very much, @michaelgrahamevans!

**Pull request merged**